### ISSUE TITLE
Problem to import files

### ISSUE THREAD

**Issue by EvandroLG** [luarocks/luarocks#1183]
I just updated one of my packages on luarocks. Locally it's everything working fine, but it's not working well on luarocks. Once I try to import the package, it's breaking because it can not find the project files. I'm getting this error:

`/usr/local/share/lua/5.3/simple_test.lua:1: module './assertions' not found`

I've this piece of code in the rockspec file:
```
build = {
  type = "builtin",
  modules = {
    ["simple_test"] = "src/simple_test.lua",
    ["simple_test.utils"] = "src/utils.lua",
    ["simple_test.assertions"] = "src/assertions.lua",
    ["simple_test.colors"] = "src/colors.lua"
  }
}
```

And in my code, I'm importing as follows:
```
local assertions = require('./assertions')
```

I did something wrong?

Follow is my rockspec:
https://github.com/EvandroLG/simple_test/blob/master/simple_test-0.0.8-0.rockspec

Thank you,
Evandro

**Comment by hishamhm** [comment]
> local assertions = require('./assertions')

Avoid using paths in require: use module names instead, and let the Lua package path figure it out. The name you use in `modules` is the name you need to use in `require`:

```
local assertions = require('simple_test.assertions')
```

Note that this will make it work with the LuaRocks-installed package but it won't work locally from the plain repo because your folder structure doesn't match the module structure. I recommend making it match, e.g.:

```
    ["simple_test.assertions"] = "src/simple_test/assertions.lua",
```

and then add `./src/?.lua` to your `LUA_PATH` variable (or `package.path` inside Lua).

**Comment by EvandroLG** [comment]
Cool!
Thanks!

I'm wondering why my original approach works well in another project. -https://github.com/EvandroLG/array.lua

If you check it, I'm using relative paths as well and the rockspec contains the following module structure:

```
modules = {
    ["array"] = "src/array.lua",
    ["array.utils"] = "src/utils.lua"
  }
```

Can you clarify why is it working well in the other project, please?

Thank you,
Evandro

**Comment by hishamhm** [comment]
Impossible to tell, without knowing your Lua paths and where every file is on your filesystem. :) You might happen to have the files locally in the right places that happen to match, or a package.path that adjusts the paths in a way that cause them to match. 

Some key facts:

* Lua itself has no understanding of directories, so it will concatenate whatever is in those names. So by using `require("./utils")` and `/usr/local/share/lua/5.3/?.lua` in your `package.path` you end up with `/usr/local/share/lua/5.3/./utils.lua`.
* Lua builds paths based only on the `package.path` and _not_ on the directory of the caller module, so modules are not "relative to each other" if you use `./` in `require`. It might, however, take the _current working directory_ (which may or may not have anything to do with the location of the caller module) if your package.path happens to include `./?.lua`, which the default package.path does.
* whereas LuaRocks installs modules based on the module names, matching them to a directory structure that causes `require` to work. So for `array.utils`, in order to make `require("array.utils")` to work, for a tree on `/usr/local/` it will be installed at `/usr/local/share/lua/array/utils.lua`.

tl;dr: always use in `require()` the same names you use in the keys of `build.modules` in the rockspec.

Hope this helps!
